python/ml_wrappers/model/predictions_wrapper.py

```diff
@@ -60,7 +61,18 @@ def _get_filtered_data(
                     data_copy = data_copy[data_copy[column_name].isna()]
                 else:
                     data_copy = data_copy[data_copy[column_name] == column_data]
-            filtered_df = data_copy
+
+                if len(data_copy) == 1 or len(data_copy) == 0:
+                    # Stop the search if we have found a possible exact match or no match
+                    break
+
+            if len(data_copy) == 1:
```

Comment:
nit: maybe this could be simplified to:
```
create_new_dataframe = len(data_copy) == 1 && !query_test_data_row.equals(
    data_copy.filter(self._feature_names))
if create_new_dataframe:
    filtered_df = pd.DataFrame(columns=self._combined_data.columns)
else:
    filtered_df = data_copy
```

Reply:
Thanks for the suggestion. I will the logic as is right now.